### PR TITLE
Domestic payments v4

### PR DESCRIPTION
**Pull request by mariantiris**
Issue: https://github.com/SecureApiGateway/SecureApiGateway/issues/1558

- Refactored packages based on datamodel changes
- Implemented support for domestic payments v4

### CODE REVIEW

**Comment by dbadham-fr** [comment]
Can you refactor the consent client to follow the structure of domestic standing orders please?

i.e. follow this structure:
image

**Comment by mariantiris** [comment]
> Can you refactor the consent client to follow the structure of domestic standing orders please?
> 
> i.e. follow this structure: image

Updated just now to follow the suggested pattern example.

File: secure-api-gateway-ob-uk-rcs-consent-store/secure-api-gateway-ob-uk-rcs-consent-store-client/src/main/java/com/forgerock/sapi/gateway/rcs/consent/store/client/payment/domestic/BaseRestDomesticPaymentConsentStoreClient.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+ */
+public class BaseRestDomesticPaymentConsentStoreClient extends BaseRestConsentStoreClient implements DomesticPaymentConsentStoreClient {
+
+    private final String consentServiceBaseUrl;
+
+    @Autowired
+    public BaseRestDomesticPaymentConsentStoreClient(ConsentStoreClientConfiguration consentStoreClientConfiguration, RestTemplateBuilder restTemplateBuilder,
+                                                 ObjectMapper objectMapper) {
+        this(consentStoreClientConfiguration, restTemplateBuilder, objectMapper, OBVersion.v3_1_10);
+    }
```
Comment (dbadham-fr) [inline secure-api-gateway-ob-uk-rcs-consent-store/secure-api-gateway-ob-uk-rcs-consent-store-client/src/main/java/com/forgerock/sapi/gateway/rcs/consent/store/client/payment/domestic/BaseRestDomesticPaymentConsentStoreClient.java:46]: Remove this constructor

Comment (mariantiris) [reply secure-api-gateway-ob-uk-rcs-consent-store/secure-api-gateway-ob-uk-rcs-consent-store-client/src/main/java/com/forgerock/sapi/gateway/rcs/consent/store/client/payment/domestic/BaseRestDomesticPaymentConsentStoreClient.java]: Done

File: secure-api-gateway-ob-uk-rcs-consent-store/secure-api-gateway-ob-uk-rcs-consent-store-client/src/main/java/com/forgerock/sapi/gateway/rcs/consent/store/client/payment/domestic/BaseRestDomesticPaymentConsentStoreClient.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+import com.forgerock.sapi.gateway.uk.common.shared.api.meta.obie.OBVersion;
+
+/**
+ * Implementation of the DomesticPaymentConsentStoreClient which makes REST calls over HTTP
+ */
+public class BaseRestDomesticPaymentConsentStoreClient extends BaseRestConsentStoreClient implements DomesticPaymentConsentStoreClient {
+
+    private final String consentServiceBaseUrl;
+
+    @Autowired
```
Comment (dbadham-fr) [inline secure-api-gateway-ob-uk-rcs-consent-store/secure-api-gateway-ob-uk-rcs-consent-store-client/src/main/java/com/forgerock/sapi/gateway/rcs/consent/store/client/payment/domestic/BaseRestDomesticPaymentConsentStoreClient.java:41]: Remove `@Autowired` annotation, as this class is not a Spring bean. 

The subclasses are beans (annotated with `@Component`) and their constructors may be autowired.

Comment (mariantiris) [reply secure-api-gateway-ob-uk-rcs-consent-store/secure-api-gateway-ob-uk-rcs-consent-store-client/src/main/java/com/forgerock/sapi/gateway/rcs/consent/store/client/payment/domestic/BaseRestDomesticPaymentConsentStoreClient.java]: removed

**Review by dbadham-fr — DISMISSED** [review]
Thanks

**Comment by mariantiris** [comment]
@dbadham-fr thanks for the review, will need re-approval as I pushed pom again to prepare it for merging with the main branch

**Review by dbadham-fr — APPROVED** [review]

**Pull request merged**